rstar/src/point.rs

```diff
@@ -344,3 +346,90 @@ implement_point_for_array!(0, 1, 2, 3, 4, 5);
 implement_point_for_array!(0, 1, 2, 3, 4, 5, 6);
 implement_point_for_array!(0, 1, 2, 3, 4, 5, 6, 7);
 implement_point_for_array!(0, 1, 2, 3, 4, 5, 6, 7, 8);
+
+macro_rules! fixed_type {
+    ($expr:expr, $type:ty) => {
+        $type
+    };
+}
+
+macro_rules! impl_point_for_tuple {
+    ($($index:expr => $name:ident),+) => {
+        impl<S> Point for ($(fixed_type!($index, S),)+)
+        where
+            S: RTreeNum
+        {
+            type Scalar = S;
+
+            const DIMENSIONS: usize = count_exprs!($($index),*);
+
+            fn generate(generator: impl Fn(usize) -> S) -> Self {
```

Comment:
I think this should indeed be
```rust
fn generate(generator: impl FnMut(usize) -> S) -> Self;
```
as in the [trait definition](https://github.com/georust/rstar/blob/62f44f3f649eda46efe0c6a96c90b0e8a9e7586d/rstar/src/point.rs#L169) to fix the build errors.